frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -1090,6 +1112,73 @@ function ReaderBookmark:setBookmarkNote(item, from_highlight, is_new_note, new_t
     self.input:onShowKeyboard()
 end
 
+function ReaderBookmark:editHighlightedText(item)
+    local input_dialog
+    input_dialog = InputDialog:new{
+        title = _("Edit highlighted text"),
+        description = "   " .. self._getDialogHeader(item),
+        input = item.notes,
+        allow_newline = true,
+        add_scroll_buttons = true,
+        use_available_height = true,
+        buttons = {
+            {
+                {
+                    text = _("Cancel"),
+                    id = "close",
+                    callback = function()
+                        UIManager:close(input_dialog)
+                    end,
+                },
+                {
+                    text = _("Save"),
+                    is_enter_default = true,
+                    callback = function()
+                        self:setHighlightedText(item, input_dialog:getInputText())
+                        UIManager:close(input_dialog)
+                    end,
+                },
+            }
+        },
+    }
+    UIManager:show(input_dialog)
+    input_dialog:onShowKeyboard()
+end
+
+function ReaderBookmark:setHighlightedText(item, text)
+    local edited
+    if text then
+        edited = true
+    else -- reset to selected text
+        if self.ui.rolling then
+            text = self.ui.document:getTextFromXPointers(item.pos0, item.pos1)
+        else
+            text = self.ui.document:getTextFromPositions(item.pos0, item.pos1).text
+        end
+    end
+    -- highlight
+    local hl = self.ui.highlight:getHighlightByDatetime(item.datetime)
+    hl.text = text
+    hl.edited = edited
+    -- bookmark
+    local index = self:getBookmarkIndexBinarySearch(item) or self:getBookmarkIndexFullScan(item)
+    local bm = self.bookmarks[index]
+    local is_auto_text_before = self:isBookmarkAutoText(bm)
+    bm.notes = text
+    if is_auto_text_before then
+        bm.text = self:getBookmarkAutoText(bm)
+    end
+    bm.edited = edited
```

Comment:
Less obvious with our crappy named bookmark properties :/
`text_edited` but it's not the `text` property, it's the `notes` property, so `notes_edited` ? But that just increase the confusion :/

Reply:
That's why I denied `text_edited`.
Also would like to avoid long `highlighted_text_edited`.
Also would like to avoid different key names in bm and hl.